src/Components/ExtractLevelsMenu.jsx

```diff
@@ -0,0 +1,184 @@
+
+import React, {useState, useEffect} from 'react'
+import Menu from '@mui/material/Menu'
+import MenuItem from '@mui/material/MenuItem'
+import {makeStyles} from '@mui/styles'
+import {TooltipIconButton} from './Buttons'
+import useStore from '../store/useStore'
+import {getModelCenter} from '../utils/cutPlane'
+import {addHashParams, getHashParams} from '../utils/location'
+import {Vector3} from 'three'
+import {useLocation} from 'react-router-dom'
+import {removePlanes} from '../utils/cutPlane'
+import LevelsIcon from '../assets/2D_Icons/Levels.svg'
+import PlanViewIcon from '../assets/2D_Icons/PlanView.svg'
+import {extractHeight} from '../utils/extractHeight'
+
+/**
+ * BasicMenu used when there are several option behind UI button
+ * show/hide from the right of the screen.
+ *
+ * @param {Array} listOfOptions Title for the drawer
+ * @return {object} ItemPropertiesDrawer react component
+ */
+export default function ExtractLevelsMenu({listOfOptions, icon, title}) {
+  const [anchorEl, setAnchorEl] = useState(null)
+  const classes = useStyles()
+  const open = Boolean(anchorEl)
+
+  const model = useStore((state) => state.modelStore)
+  let [floorplanMenuItems, showExtractMenu] = useState([])
+
+
+  const PLANE_PREFIX = 'p'
+  const handleClick = (event) => {
+    setAnchorEl(event.currentTarget)
+  }
+  const handleClose = () => {
+    setAnchorEl(null)
+  }
+  const viewer = useStore((state) => state.viewerStore)
+  const location = useLocation()
+  const createFloorplanPlane = (h1, h2) => {
+    console.log('Got here')
+    removePlanes(viewer)
+    const modelCenter1 = new Vector3(0, h1, 0)
+    const modelCenter2 = new Vector3(0, h2, 0)
+    const normal1 = new Vector3(0, 1, 0)
+    const normal2 = new Vector3(0, -1, 0)
+    viewer.clipper.createFromNormalAndCoplanarPoint(normal1, modelCenter1)
+    viewer.clipper.createFromNormalAndCoplanarPoint(normal2, modelCenter2)
+  }
+  const planView = () => {
+    viewer.context.ifcCamera.toggleProjection()
+    viewer.plans.moveCameraTo2DPlanPosition(true)
+    const yConst = 100 // value used in moveCameraTo2DPlanPosition in web-ifc
+    const modelCenterX = getModelCenter(model).x
+    const modelCenterY = getModelCenter(model).y
+    const modelCenterZ = getModelCenter(model).z
+    viewer.context.ifcCamera.cameraControls.setLookAt(modelCenterX, yConst, modelCenterZ, modelCenterX, 0, modelCenterZ, true)
```

Comment:
for this and later lines, use an intermediate variable for viewer.context.ifcCamera.
```
const camera = viewer.context.ifcCamera
camera.cameraControls...
camera.projectionManager...
```

Reply:
Fixed